File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = strs[0];
        for(int i=1; i<strs.size(); i++){
            string s = strs[i];
            string temp;
            int n = min(s.length(), prefix.length());
            for(int j=0; j<n; j++){
                if (s[j] == prefix[j]) temp += s[j];
                else break;
            }
            prefix = temp;
        }

        return prefix;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    l_pref := []byte(strs[0])
    for i := 1; i < len(strs); i++ {
        str := []byte(strs[i])
        j := 0
        for ; j < min(len(str), len(l_pref)); j++ {
            if str[j] != l_pref[j] {
                break
            }
        }

        l_pref = str[:j]
    }

    return string(l_pref)
}